spec/acceptance/api/account_spec.rb

```diff
@@ -14,6 +14,8 @@
     account
   end
 
+  # FIXME: admin_display_name should be always present, test should pass?
+  # let(:expected_provider_fields) { %w[admin_domain domain admin_base_url base_url from_email support_email finance_support_email site_access_code admin_display_name] }
```

Comment:
Problems with this:
1. This variable is used for testing both the JSON and the XML responses, but for the JSON it is using the AccountRepresenter which you edited, as for the XML, it is using the Account#to_xml method. To make it work for XML, you should add to that method something like `xml.admin_display_name display_name` or `xml.admin_display_name first_admin&.display_name` or something like that...
2. We have 2 different concepts for "display name" of an Account now, which is confusing like hell... you want the `first_admin.display_name`, which is:
https://github.com/3scale/porta/blob/acc77228aa4f1d1ac226c78cae0e421588de6990/app/models/user.rb#L317-L320
And, the other concept of Account#display_name is the one in the model which does not return the same thing (it uses the `username` of the user and never its name):
https://github.com/3scale/porta/blob/acc77228aa4f1d1ac226c78cae0e421588de6990/app/models/account.rb#L361-L364
3. What `expected_provider_fields` is used for below is to do this (simplified):

- For **JSON**: `should have_properties('admin_display_name').from(resource)`. In here:
https://github.com/3scale/porta/blob/acc77228aa4f1d1ac226c78cae0e421588de6990/spec/acceptance/api/account_spec.rb#L168

- For **XML**: `it { should have_tags('admin_display_name').from(resource) }` In here:
https://github.com/3scale/porta/blob/acc77228aa4f1d1ac226c78cae0e421588de6990/spec/acceptance/api/account_spec.rb#L218

What this does internally is [here](https://github.com/3scale/porta/blob/3122cd7781c421b330433a413b9ca54ea665ec2e/spec/support/api/matchers.rb#L1-L46), but as a simplified idea, this is what it does:
`assert resource.public_send(attribute) == json[attribute]` (`resource` being the model of account in this case). `resource.public_send(attribute)` cannot work for  `admin_display_name` because there isn't a method with that name inside this model

Reply:
I have a few possible solutions in mind but I do not like any 😓 : 
1. Remove the `delegate :display_name, to: :first_admin, allow_nil: true` line to use the `display_name` of `Account`, just as it is, without changing it. This means that you would always have the `username`, and never the `name` of the user.
2. Option 1 but changing the `Account` model `display_name` to return always the `display_name` of the user instead of the `username`.
3. Keep the `display_name` as it is in the models, remove the `delegate :display_name, to: :first_admin, allow_nil: true` from the representer, and add a new `admin_display_name` that returns actually what u want

Reply:
We don't care about XML, we should deprecate it: Old API who have XML still works but newer APIs or response might not be compatible

Reply:
That's not the biggest issue at all IMO 😄 that it solved in the test by not adding the `expected_provider_fields` and instead just doing directly `it { should have_properties('admin_display_name').from(resource) }` or even directly an equality comparison there.
But I've mentioned much more above that should be addressed. And all my proposed solutions above have drawbacks.

Reply:
`display_name` should never be on the model (nothing to do with database) but in the representer

And if there is conflict with `user.display_name` or anything else then we should address it already